src/starknet-types.ts

```diff
@@ -102,6 +102,27 @@ export interface Transaction {
     transaction_index: number;
 }
 
+export interface FunctionInvocation {
+    call_type: string;
+    calldata: string[];
+    caller_address: string;
+    class_hash: string;
+    contract_address: string;
+    entry_point_type: string;
+    events: unknown[];
+    execution_resources: ExecutionResources;
+    internal_calls: unknown[];
+    messages: unknown[];
+    result: string[];
+    selector: string;
+}
+
+export interface TransactionTrace {
+    function_invocation: FunctionInvocation;
```

Comment:
Some of these could be made optional (`?`) if they aren't present in all types of txs.

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/264/commits/583dac4fa6885dabb6f19d31171947d5d9f70e6e